parser/src/statements/try_catch_statement.rs

```diff
@@ -1 +1,59 @@
+use std::borrow::Cow;
 
+use crate::{ASTNode, Block, TSXKeyword, TSXToken, VariableIdentifier};
+use source_map::Span;
+use visitable_derive::Visitable;
+
+#[derive(Debug, PartialEq, Eq, Clone, Visitable)]
+#[cfg_attr(feature = "self-rust-tokenize", derive(self_rust_tokenize::SelfRustTokenize))]
+pub struct TryCatchStatement {
+	pub inner: Block, // both try and catch blocks must always be braced
+	pub error_var: VariableIdentifier,
```

Comment:
Just checked the spec and there are a few things additional things around `error_val`
- It can be missing entirely (parenthesis and all)
- It can include destructuring `catch ({a, b})`, so can use [VariableField](https://docs.rs/ezno-parser/0.0.3/ezno_parser/enum.VariableField.html) which adds the additional destructuring variants
- [TypeScript allows an optional annotation here](https://stackoverflow.com/a/69577476/10048799), so can use [TypeReference](https://docs.rs/ezno-parser/0.0.3/ezno_parser/types/type_references/enum.TypeReference.html)
So `error_val: Option<(VariableField, Option<TypeReference>)>` would be great